psi4/src/psi4/libscf_solver/frac.cc

```diff
@@ -207,18 +186,26 @@ void HF::frac_renormalize() {
     for (int ind = 0; ind < options_["FRAC_OCC"].size(); ind++) {
         int i = options_["FRAC_OCC"][ind].to_integer();
         double val = options_["FRAC_VAL"][ind].to_double();
+        if (denom) { val = 1.0 / val; }
         bool is_alpha = (i > 0);
         i = std::abs(i) - 1;  // Back to C ordering
 
-        int h = ((is_alpha) ? std::get<1>(pairs_a[i]) : std::get<1>(pairs_b[i]));
+        int h; double** Cp; int j;
+        if (is_alpha) {
+            h = std::get<1>(pairs_a[i]);
+            j = std::get<2>(pairs_a[i]);
+            Cp = Ca_->pointer(h);
+        } else {
+            h = std::get<1>(pairs_b[i]);
+            j = std::get<2>(pairs_b[i]);
+            Cp = Cb_->pointer(h);
+        }
 
         int nso = Ca_->rowspi()[h];
         int nmo = Ca_->colspi()[h];
 
-        double** Cp = ((is_alpha) ? Ca_->pointer(h) : Cb_->pointer(h));
-
-        // And I say all that to say this: TODO: This destroys FMP2 computations if val == 0
-        if (val != 0.0) C_DSCAL(nso, 1.0 / std::sqrt(val), &Cp[0][i], nmo);
+        // And I say all that to say this:
+        C_DSCAL(nso, std::sqrt(val), &Cp[0][j], nmo);
```

Comment:
the `[i]` -> `[j]` was the indexing error? explains why I can't make the code equivalent.

Reply:
Yes. The old code used the letter "i" earlier in the code to refer to the index of the orbital within the irrep, but at this point in the code, "i" means something else entirely. The fix is to save the index of the orbital to a new variable, which I call "j".